functions/dialect.go

```diff
@@ -85,13 +83,14 @@ type withID interface {
 	ID() extensions.ID
 }
 
-func makeLocalFunctionVariantMap[T withID, V any](dialectFunctionInfos map[extensions.ID]*dialectFunctionInfo, getFunctionVariants func(string) []T, createLocalVariant func(T, *dialectFunctionInfo) *V) (map[FunctionName][]*V, error) {
+func makeLocalFunctionVariantMap[T withID, V any](dialectFunctionInfos map[extensions.ID]*dialectFunctionInfo, getFunctionVariants func(string) []T, createLocalVariant func(T, *dialectFunctionInfo) *V) (map[FunctionName][]*V, []*V, error) {
```

Comment:
It'd be worth documenting this function's outputs.  It appears we are returning all local variants as a side effect of making a map of local function variants.  If they're the same data but represented two ways, we could use a new combined type (named a MapSlice or something local function specific) so we are only returning the data one way.